.github/workflows/kubernetes-compute-cpu-resources-cleanup.yml

```diff
@@ -58,36 +53,27 @@ jobs:
 
     # cleanup resources
     - name: delete_endpoints
-      if: ${{ always() }}
+      if: ${{ always() && github.event.inputs.DELETE_ENDPOINTS == 'true' }}
```

Comment:
do we still need always?

Reply:
yes. always() means ignoring the status of previous steps. and I also set separate github.event.inputs.DELETE_ENDPOINTS flag for each step to manually control which step to run.